security/pkg/workload/server.go

```diff
@@ -190,16 +180,35 @@ func (s *SDSServer) RegisterUdsPath(udsPath string) error {
 		return fmt.Errorf("failed to listen on %v", err)
 	}
 
+	var opts []grpc.ServerOption
+	udsServer := grpc.NewServer(opts...)
+	sds.RegisterSecretDiscoveryServiceServer(udsServer, s)
+	s.udsServerMap[udsPath] = udsServer
+
 	// grpcServer.Serve() is a blocking call, so run it in a goroutine.
 	go func() {
-		log.Infof("Starting GRPC server on UDS path", udsPath)
-
-		err := s.grpcServer.Serve(listener)
+		log.Infof("Starting GRPC server on UDS path: %s", udsPath)
+		err := udsServer.Serve(listener)
 		// grpcServer.Serve() always returns a non-nil error.
 		log.Warnf("GRPC server returns an error: %v", err)
```

Comment:
Add a TODO here to pass the error back to the caller, so that correct operations can be conducted.